cli/join.go

```diff
@@ -42,11 +42,10 @@ func join(c *cli.Context) {
 	}
 
 	hbval := time.Duration(hb)
+	log.WithFields(log.Fields{"addr": addr, "discovery": dflag}).Infof("The entry in the discovery will disappear if the node goes down..", hbval)
+	if err := d.Register(addr); err != nil {
+		log.Error(err)
+	}
 	for {
```

Comment:
Actually you were right. I would keep this the way it was and continue issuing periodic `Register()` in case something goes wrong (instead of doing just one Register)